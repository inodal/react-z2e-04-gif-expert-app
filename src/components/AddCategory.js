import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        // Avoid default behaviour that will submit and refresh whole view
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue.trim(), ...cats]); // Show new first
            setInputValue('');
        }
        
    }

    return (
        <>
        <form onSubmit= { handleSubmit }>
        <p>{ inputValue }</p> 
            <input 
                type ="text"
                value = { inputValue }
                onChange = { handleInputChange }
                
            />
        </form>
            
        </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}

