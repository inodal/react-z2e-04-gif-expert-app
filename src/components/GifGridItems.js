import React from 'react'

export const GifGridItems = ({id, title, url }) => {

    console.log(`${id}:\ntitle: ${title} \nurl: ${url}`);
    return (
        <div className="card">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}
