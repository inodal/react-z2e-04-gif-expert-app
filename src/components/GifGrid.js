import React, { useState, useEffect } from 'react'
import { GifGridItems } from './GifGridItems';

export const GifGrid = ({category}) => {

    
    const [images, setImages] = useState([]);

    // Use hook 'useEffect' to render only the first time
    useEffect(() => {
        getGifs()
    }, [])

    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?&limit=10&q=Dragon+Ball&api_key=a6moXgoQ6vuFxhVHum2nE6CPtZdV66Uz';
        const resp = await fetch(url);
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }

        });
        console.log(gifs);
        setImages(gifs);
    };

    return (
        <div>
            <h3> { category } </h3>
            {
                images.map(img => (
                    <GifGridItems 
                        key={ img.id }
                        { ...img } 
                    />
                ))
            }
        </div>
    )
}
