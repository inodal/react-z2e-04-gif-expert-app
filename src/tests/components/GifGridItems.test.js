import React from 'react'
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { GifGridItems } from "../../components/GifGridItems";

describe('GifGridItems test suite', () => {
    test('should show GifGridItems component properly', () => {
        const data = {
            data: [],
            loading: true
        };
        // const wrapper = shallow(<GifGridItems key={1} {...data} />);
        const wrapper = shallow(<GifGridItems />);

        expect(wrapper).toMatchSnapshot();
    })
    
})
