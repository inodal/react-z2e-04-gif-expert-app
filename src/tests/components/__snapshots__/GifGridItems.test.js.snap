// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItems test suite should show GifGridItems component properly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
